File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = {}
        for num in nums:
            if freqs.get(num) == None:
                freqs[num] = 1
            else:
                freqs[num] += 1
        sortedFreqs = dict(sorted(freqs.items(), key = lambda item: item[1], reverse = True))
        results = [key for key, value in sortedFreqs.items()]
        return results[:k]

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    class KV<K, V extends Comparable<V>> implements Comparable<KV<K, V>> {
        K key;
        V value;

        KV(K key, V value) {
            this.key = key;
            this.value = value;
        }

        @Override
        public int compareTo(KV<K, V> other) {
            return value.compareTo(other.value);
        }
    }

    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap();
        for (int num : nums) {
            if (map.containsKey(num)) {
                map.put(num, map.get(num) + 1);
            } else {
                map.put(num, 1);
            }
        }

        KV<Integer, Integer>[] frequences = new KV[map.size()];
        int i = 0;
        for (Map.Entry<Integer, Integer> e : map.entrySet()) {
            frequences[i] = new KV(e.getKey(), e.getValue());
            i++;
        }   

        Arrays.sort(frequences);

        int[] result = new int[k];
        for (i = 0; i < k; i++) {
            result[i] = frequences[frequences.length - i -1].key;
        }

        return result;
    }
}